[docs] Describe `--cache` (#5969)
DevExpress/testcafe: docs/articles/documentation/guides/advanced-guides/speed-up-test-execution.md

This approach ensures that the page model object is created only once per test run.

> Important! If you export the page model class and create an instance in every test file, JavaScript creates a new page object for each test file. This increases test execution time and memory consumption.

For more on page models, read [Page Model](../../guides/concepts/page-model.md).

## Enable Server-Side Caching

> Important! Support for server-side caching is **experimental**.
>

> > Don't use caching if you run into compatibility issues with your tests.

Reviewer: ```suggestion
> Do not use caching if you run into compatibility issues with your tests.
```

> > Important! If you export the page model class and create an instance in every test file, JavaScript creates a new page object per test file. This increases test execution time and memory consumption.

Reviewer: ```suggestion
> Important! If you export the page model class and create an instance in every test file, JavaScript creates a new page object for each test file. This increases test execution time and memory consumption.
```

For more on page models, read [Page Model](../../guides/concepts/page-model.md).

## Enable Server-Side Caching

> > Important! Server-side caching is **experimental**.

Reviewer 2: Does we need a predicate here?

`Important! Server-side caching is **experimental** FEATURE`.

Author: No, this sentence is correct.

Reviewer 2: Ok

>

> > Disable caching if you run into compatibility issues with your tests.

Reviewer 2: I propose to rephrase this sentence since the feature is disabled by default.

`Don't use caching if you run into compatibility issues...`

> The TestCafe proxy can cache webpage assets (such as stylesheets, scripts, and images) and retrieve them from its cache the next time it accesses the webpage.

Reviewer: ```suggestion
The TestCafe proxy can cache webpage assets (stylesheets, scripts, images) and retrieve them from its cache when it accesses the webpage again.
```

> TestCafe can save webpage assets (like styles, scripts, and images) in its server cache. When TestCafe visits the page again, it pulls assets from the cache instead of requesting them from server.

Reviewer 2: `TestCafe can store...` is better variant for this.

> TestCafe can store webpage assets (like styles, scripts, and images) in its server cache. When TestCafe visits the page again, it pulls assets from the cache instead of requesting them from server.

Reviewer 3: + TestCafe doesn't visit pages, the browser does
+ 'server cache' and 'server-side cache' are two different things, and the former expression can be misinterpreted

```suggestion
The TestCafe proxy can cache webpage assets (such as stylesheets, scripts, and images) and retrieve them from its cache the next time it accesses the webpage.
```

You can enable caching in one of the following ways:

* [--cache](../../reference/command-line-interface.md#cache) CLI option

  ```sh
  testcafe chrome my-tests/ --cache
  ```

* [cache]() configuration file property

  ```json
  {
  "cache": true
  }
  ```

* [cache]() API option

  ```js
  runner.run({ cache: true });
  ```

> Enable server-side caching when you test asset-heavy webpages to decrease test run time.

Reviewer 2: `you test asset-heavy webpages to decrease test run time` 
It's difficult to read. Let's try to describe in more detail.
The `cache` feature can useful if your web application loads a lot assets (scripts, styles and etc.) and browser should cache these assets.

Author: > If the tested application loads assets that are cached by the browser, enable server-side caching to decrease test run time.

Reviewer 2: Let's emphasize that there are many assets:

```js
If the tested application loads A LOT OF assets...
```

Author: `a lot` is not very good for docs because it leaves the user wondering "how much is that exactly?" On the other hand, we can't give a specific number here because it would be arbitrary.

`asset-heavy webpages` means `applications that load a lot of assets`

I propose the following:

> To decrease test run time, enable `--cache` when you test asset-heavy web applications.

> Enable server-side caching to decrease test run time, particularly when your application relies on browser caching.

Reviewer: ```suggestion
Enable server-side caching to decrease test duration, particularly when your application relies on browser caching.
```

> If the tested application loads assets that are cached by the browser, enable server-side caching to decrease test run time.

Reviewer 3: lots of issues with this sentence. it doesn't do a good job of explaining the rationale behind the feature. did I get it right?
```suggestion
Enable server-side caching to decrease test run time, particularly when you expect your users to rely on browser caching.
```

Author: Users can't rely on caching, the app can

```suggestion
Enable server-side caching to decrease test run time, particularly when your application relies on browser caching.
```
